sexp/src/main/java/com/novoda/sexp/Streamer.java

```diff
@@ -0,0 +1,41 @@
+package com.novoda.sexp;
+
+import com.novoda.sax.RootElement;
+
+/**
+ * This streamer can be used when you want to return your parsed object synchronously.
+ * A good example being {@link SimpleEasyXmlParser#parse(String, Streamer)}
+ *
+ * @param <R> the type of object you expect as a result of xml parsing
+ */
+public interface Streamer<R> {
+    /**
+     * @return The root tag of your XML file
+     */
+    RootTag getRootTag();
+
+    /**
+     * This is where you stream the xml 'tree' by using
+     * {@link com.novoda.sexp.finder.ElementFinder ElementFinder}
+     * or {@link com.novoda.sexp.parser.Parser Parser}
+     * objects to parse the XML.<br/><br/>
+     * You can use the {@link com.novoda.sexp.finder.ElementFinderFactory ElementFinderFactory}
+     * to create tree crawlers, or you can create your own if you implement one of:
+     * <ul>
+     * <li>{@link com.novoda.sexp.finder.ElementFinder ElementFinder}</li>
+     * <li>{@link com.novoda.sexp.marshaller.AttributeMarshaller AttributeMarshaller}</li>
+     * <li>{@link com.novoda.sexp.marshaller.BodyMarshaller BodyMarshaller}</li>
+     * <li>{@link com.novoda.sexp.parser.Parser Parser}</li>
+     * <li>{@link com.novoda.sexp.parser.ParseWatcher ParseWatcher}</li>
+     * <li>{@link com.novoda.sexp.parser.ListParseWatcher ListParseWatcher}</li>
+     * </ul>
+     *
+     * @param rootElement the root element of your XML file
+     */
+    void stream(RootElement rootElement);
```

Comment:
We based our bet for this name after reading the following http://www.extreme.indiana.edu/xgws/xsoap/xpp/

> Xml Pull Parser (in short XPP) is a streaming pull XML parser and should be used when there is a need to process quickly and efficiently all input elements (for example in SOAP processors). Pull parsing is just one level up from tokenizing XML (and that is actually how XPP2 default implementation is done).